extras/src/test/java/com/google/gson/typeadapters/RuntimeTypeAdapterFactoryTest.java

```diff
@@ -167,6 +167,25 @@ public void testSerializeCollidingTypeFieldName() {
     }
   }
 
+  public void testSerializeWrappedNullValue() {
+    TypeAdapterFactory billingAdapter = RuntimeTypeAdapterFactory.of(BillingInstrument.class)
```

Comment:
RuntimeTypeAdapterFactory is not in Gson's public API. Can you demonstrate this problem using the public API only?